micro-rdk/src/common/webrtc/api.rs

```diff
@@ -505,7 +528,18 @@ where
 
         // TODO use is_some_then when rust min version reach 1.70
         if current_prio >= caller_prio {
-            return Err(WebRtcError::CurrentConnectionHigherPrority());
+            let sdp = WebRtcSdp::new(sess_desc, offer.uuid.clone());
+            let sig_channel = self
+                .signaling
+                .as_mut()
+                .ok_or(WebRtcError::SignalingDisconnected())?;
+
+            sig_channel.send_sdp_error(&sdp).await?;
+
+            // this delay ensures that the sdp error is properly sent and received before closing the connection.
```

Comment:
can you expand on this? did we confirm 100% it's a race with writing to the MongoDB document?

Reply:
Not confirmed yet. Should know tomorrow and will attach a ticket if needed. Wanted to get a first pass done while I wrap that up.